spinalcordtoolbox/image.py

```diff
@@ -287,10 +288,12 @@ def __init__(self, param=None, hdr=None, orientation=None, absolutepath=None, di
         elif isinstance(param, list):
             self.data = np.zeros(param)
             self.hdr = hdr
+            self.hdr.set_data_shape(self.data.shape)
         # create a copy of im_ref
         elif isinstance(param, (np.ndarray, np.generic)):
             self.data = param
             self.hdr = hdr
+            self.hdr.set_data_shape(self.data.shape)
```

Comment:
NB: This change is a bugfix (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/commits/230b522355d017d8c874b75d96b704985d08ce71) and could possibly be merged outside of this PR.

Reply:
> These `hdr` objects are pretty small, so a quick, inexpensive fix would be to make a copy before assigning to `self.hdr`.

I very much agree here. In fact, I'm surprised we weren't making a copy in the first place, given that we do so in `loadFromPath` while loading images, presumably for the exact reasons you describe:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b4cf266565ca46de65f597f43292a328acd4eec4/spinalcordtoolbox/image.py#L411-L415

Reply:
I'm realizing that the `self.hdr = hdr` statements are bit more complicated than they seem. So, I think I'm going to factor this fix out into a different PR so that it can be merged separately.

Reply:
Commit reverted in 1ccba69629e50c89e24d2e827341b9a4b4f02616, and the changes have been migrated to https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3931.